app/routines/find_or_create_unclaimed_user.rb

```diff
@@ -0,0 +1,90 @@
+# Find or create a new user with state "unclaimed"
+#
+# Given either an email address or username:
+#   attempt to find a user with that attribute.
+#   If the user is found, return the user
+#     Otherwise create a new user with the email and username given,
+#     set it's state to "unclaimed" and return that record
+
+class FindOrCreateUnclaimedUser
+
+  lev_routine
+
+  uses_routine CreateUser, translations: { outputs: { type: :verbatim } }
+  uses_routine AddEmailToUser
+  uses_routine SetPassword
+  uses_routine CreateIdentity
+
+  protected
+
+  def exec(options)
+    user = nil
+
+    if options[:username]
+      user = find_or_create_by_username(options)
+    elsif options[:email]
+      user = find_or_create_by_email(options)
+    else
+      fatal_error(code: :invalid_input, message: "Must provide either email or username")
```

Comment:
The endpoint description says an email is always required, which disagrees with this error message.